src/cppcache/include/geode/CacheStatistics.hpp

```diff
@@ -102,8 +102,8 @@ class CPPCACHE_EXPORT CacheStatistics : public SharedBase {
   virtual void setLastAccessedTime(uint32_t lat);
   virtual void setLastModifiedTime(uint32_t lmt);
 
-  volatile uint32_t m_lastAccessTime;
-  volatile uint32_t m_lastModifiedTime;
+  std::atomic<uint32_t> m_lastAccessTime;
+  std::atomic<uint32_t> m_lastModifiedTime;
```

Comment:
The original code was using atomic operations already, just from multiple platform specific sources. std::atomic gets compiled into whatever is most performant for the platform. I would guess that this is actually faster than whatever we were doing on each platform since many were function calls into another library. This code will get compiled to atomic put opcodes on most platforms (with appropriate memory synchronization between caches).